plugins/scanners/scanoss/src/main/kotlin/ScanOssConfig.kt

```diff
@@ -37,5 +37,12 @@ data class ScanOssConfig(
      * Whether to write scan results to the storage.
      */
     @OrtPluginOption(defaultValue = "true")
-    val writeToStorage: Boolean
+    val writeToStorage: Boolean,
+
+    /**
+     * Whether to enable path obfuscation when sending file paths to the SCANOSS server.
+     * When enabled, the actual file paths will be obfuscated in the requests to protect sensitive information.
+     */
+    @OrtPluginOption(defaultValue = "false")
```

Comment:
@nnobelis, should this default to `true` to resemble the original behavior in ORT?

Reply:
I believe @nnobelis is unavailable today. So if you want to proceed today @isasmendiagus, then I'd go with the conservative approach and default this to "true".

Reply:
Yes, it should IMO default to `true`.

Reply:
I am not sure about these rare scenarios but we could also define a default at Bosch.

@sschuberth  What is your take on this ?

Reply:
We are happy with whatever decision you take

Reply:
> Before this PR, there was no path obfuscation. Therefore if the default is `false`, nothing has changed ?

I'm comparing to the state which originally had "always on" path obfuscation. IMO it does not make sense to compare to the (intermediate) state which had path obfuscation temporarily removed.

Reply:
See my comment [above](https://github.com/oss-review-toolkit/ort/pull/10376#discussion_r2115300327):

> But then the commit needs to be marked as breaking, and the message should highlight the change of defaults.

Reply:
@sschuberth Do you agree with this commit message?

```
feat(scanoss)!: Re-implement path obfuscation using scanoss.java 

BREAKING CHANGE: Path obfuscation default changed from true to false.
Previously, path obfuscation was configurable in ORT with default=true.
This re-implementation moves the obfuscation logic to the scanoss.java
library (v0.11.0) and changes the default to false.

Users who rely on path obfuscation must now explicitly enable it in their
configuration. 

When enabled, the scanoss.java library handles path obfuscation during
API communication while ORT preserves and returns original paths in the
final scan results.
```

Reply:
> Previously, path obfuscation was configurable in ORT with default=true.

It wasn't configurable previously, but hard-coded to always be enabled, see [above](https://github.com/oss-review-toolkit/ort/pull/10376#discussion_r2115323963).

Say: "Originally, path obfuscation was hard-coded to always be enabled."

> This re-implementation moves the obfuscation logic to the scanoss.java library (v0.11.0) and ...

The move is not something that *this* change does.

Say: "This re-implementation relies on the obfuscation logic as implemented in the scanoss.java library version 0.11.0 and ..."